### PR TITLE
enhance(main/dart): enable auto updates

### PR DESCRIPTION
**Pull request by TomJo2000**
I also took the opportunity to do some miscellaneous build script cleanup,
switching to the GitHub source also allows us to enable auto updates for dart.

> PR checklist
> - [x] Builds locally 
> - [x] Builds on CI
> - [x] Package content is complete
> - [ ] Tested on Device

<h1><em></em></h1> 

I would appreciate it if @priyanuj-gogoi could check this over,
reviews from anyone else are of course welcome as always.

### CODE REVIEW

File: packages/dart/build.sh [hunk 1]
```diff
@@ -3,31 +3,54 @@ TERMUX_PKG_DESCRIPTION="Dart is a general-purpose programming language"
...
-TERMUX_PKG_VERSION=3.4.4
-TERMUX_PKG_AUTO_UPDATE=false
+TERMUX_PKG_VERSION=3.5.3
+TERMUX_PKG_SRCURL=https://github.com/dart-lang/sdk/archive/refs/tags/${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=ee2cfea26c9eb22a5e7df5f68e13863d4e55ff14c06a286bf98d866f7b85652d
 TERMUX_PKG_BUILD_IN_SRC=true
-TERMUX_PKG_SKIP_SRC_EXTRACT=true
+TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_UPDATE_TAG_TYPE='newest-tag'
+TERMUX_PKG_UPDATE_VERSION_REGEXP='^\d+.\d+.\d+$'
```
[the remark this answers is not in the document]
Comment (Biswa96) [reply packages/dart/build.sh:12]: That check is done here.
https://github.com/termux/termux-packages/blob/388cd7e4279e8a1e154fd3af572aa79d9d72cefa/scripts/updates/utils/termux_pkg_upgrade_version.sh#L20-L32

File: packages/dart/build.sh [hunk 2]
```diff
@@ -3,31 +3,54 @@ TERMUX_PKG_DESCRIPTION="Dart is a general-purpose programming language"
...
-termux_step_get_source() {
-	mkdir -p $TERMUX_PKG_SRCDIR
-	cd $TERMUX_PKG_SRCDIR
+termux_pkg_auto_update() {
+	# Taken from packages/taplo
 
+	# Get latest release tag:
+	local api_url="https://api.github.com/repos/dart-lang/sdk/git/refs/tags"
+	local latest_refs_tags
+	latest_refs_tags=$(curl -s "${api_url}" | jq '.[].ref' | sed -ne "s|.*${TERMUX_PKG_UPDATE_VERSION_REGEXP}\"|\1|p")
```
[the remark this answers is not in the document]
Comment (truboxl) [reply packages/dart/build.sh:25]: Also caught in CI
https://github.com/termux/termux-packages/actions/runs/10939458418/job/30369751602?pr=21508#step:4:23

[the remark this answers is not in the document]
Comment (TomJo2000) [reply packages/dart/build.sh:25]: Oh crap I completely forgot about this PR.

**Comment by TomJo2000** [comment]
Closing in favor of #22145.

**Pull request closed without merging**